src/xdocs/config_coding.xml

```diff
@@ -4233,7 +4233,8 @@ abstract class AbstractExample { // OK
           protected against later changes, e.g. introduction of new types in an
           enumeration type. Note that the compiler requires switch expressions
           to be exhaustive, so this check does not enforce default branches on
-          such expressions.
+          such expressions. Switch blocks containing pattern case label elements
+          are also required to be exhaustive, and require no default label.
```

Comment:
I and probably users need more details on this.
Even original switch do not require default. Requirement for default is style or best practice concern.

Reply:
A few more details added here, and I have greatly expanded the examples section.

Reply:
Link to exact check documentation: https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/b89b1d1_2022170715/config_coding.html#MissingSwitchDefault

Reply:
>Note that the compiler requires switch expressions to be exhaustive

I do not know what it means, can you place link to more detailed explanation of exhaustive

Reply:
Updated wording and provided relevant link.